File: datasets/rs/HumanEval_70_strange_sort_list.rs
/// Given list of integers, return list in strange order.
/// Strange sorting, is when you start with the minimum value,
/// then maximum of the remaining integers, then minimum and so on.
/// Examples:
/// strange_sort_list([1, 2, 3, 4]) == [1, 4, 2, 3]
/// strange_sort_list([5, 5, 5, 5]) == [5, 5, 5, 5]
/// strange_sort_list([]) == []
fn strange_sort_list(lst: Vec<isize>) -> Vec<isize> {
    let mut lst = lst;
    let mut result = Vec::new();
    let mut min = lst.pop().unwrap();
    let mut max = lst.pop().unwrap();
    while !lst.is_empty() {
        if min < max {
            result.push(min);
            if let Some(x) = lst.pop() {
                if x < max {
                    min = x;
                } else {
                    max = x;
                }
            }
        } else {
            result.push(max);
            if let Some(x) = lst.pop() {
                if x < min {
                    max = x;
                } else {
                    min = x;
                }
            }
        }
    }
    result.push(min);
    result.push(max);
    result

}

fn main() {
    let candidate = strange_sort_list;
    assert_eq!(candidate(vec![1, 2, 3, 4]), vec![1, 4, 2, 3]);
    assert_eq!(candidate(vec![5, 6, 7, 8, 9]), vec![5, 9, 6, 8, 7]);
    assert_eq!(candidate(vec![1, 2, 3, 4, 5]), vec![1, 5, 2, 4, 3]);
    assert_eq!(candidate(vec![5, 6, 7, 8, 9, 1]), vec![1, 9, 5, 8, 6, 7]);
    assert_eq!(candidate(vec![5, 5, 5, 5]), vec![5, 5, 5, 5]);
    assert_eq!(candidate(Vec::<isize>::new()), Vec::<isize>::new());
    assert_eq!(candidate(vec![1, 2, 3, 4, 5, 6, 7, 8]), vec![1, 8, 2, 7, 3, 6, 4, 5]);
    assert_eq!(candidate(vec![0, 2, 2, 2, 5, 5, -5, -5]), vec![-5, 5, -5, 5, 0, 2, 2, 2]);
    assert_eq!(candidate(vec![111111]), vec![111111]);
}

File: datasets/java/HumanEval_70_strange_sort_list.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
class Problem {
private static <T> T getDefaultValue(Class<T> clazz) {
return (T) Array.get(Array.newInstance(clazz, 1), 0);
}    // Given list of integers, return list in strange order.
    // Strange sorting, is when you start with the minimum value,
    // then maximum of the remaining integers, then minimum and so on.
    // Examples:
    // strange_sort_list([1, 2, 3, 4]) == [1, 4, 2, 3]
    // strange_sort_list([5, 5, 5, 5]) == [5, 5, 5, 5]
    // strange_sort_list([]) == []
    public static ArrayList<Long> strange_sort_list(ArrayList<Long> lst) {
        ArrayList<Long> result = new ArrayList<Long>();
        if (lst.size() == 0) {
            return result;
        }
        long min = lst.get(0);
        long max = lst.get(0);
        for (long i : lst) {
            if (i < min) {
                min = i;
            }
            if (i > max) {
                max = i;
            }
        }
        while (result.size() < lst.size()) {
            result.add(min);
            min = Long.MAX_VALUE;
            for (long i : lst) {
                if (i > max && i < min) {
                    min = i;
                }
            }
            if (min == Long.MAX_VALUE) {
                break;
            }
            result.add(max);
            max = Long.MIN_VALUE;
            for (long i : lst) {
                if (i < max && i > min) {
                    max = i;
                }
            }
        }
        return result;
    }
    // Given list of integers, return list in strange order.
    // Strange sorting, is when you start with the minimum value,
    // then maximum of the remaining integers, then minimum and so on.
    // Examples:
    // strange_sort_list([1, 2, 3, 4]) == [1, 4, 2, 3]
    // strange_sort_list([5, 5, 5, 5]) == [5, 5, 5, 5]
    // strange_sort_list([]) == []
    public static ArrayList<Integer> strange_sort_list(ArrayList<Integer> lst) {
        ArrayList<Integer> result = new ArrayList<Integer>();
        if (lst.size() == 0) {
            return result;
        }
        int min = lst.get(0);
        int max = lst.get(0);
        for (int i : lst) {
            if (i < min) {
                min = i;
            }
            if (i

    }
    public static void main(String[] args) {
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)1l, (long)2l, (long)3l, (long)4l)))).equals((new ArrayList<Long>(Arrays.asList((long)1l, (long)4l, (long)2l, (long)3l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)5l, (long)6l, (long)7l, (long)8l, (long)9l)))).equals((new ArrayList<Long>(Arrays.asList((long)5l, (long)9l, (long)6l, (long)8l, (long)7l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)1l, (long)2l, (long)3l, (long)4l, (long)5l)))).equals((new ArrayList<Long>(Arrays.asList((long)1l, (long)5l, (long)2l, (long)4l, (long)3l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)5l, (long)6l, (long)7l, (long)8l, (long)9l, (long)1l)))).equals((new ArrayList<Long>(Arrays.asList((long)1l, (long)9l, (long)5l, (long)8l, (long)6l, (long)7l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)5l, (long)5l, (long)5l, (long)5l)))).equals((new ArrayList<Long>(Arrays.asList((long)5l, (long)5l, (long)5l, (long)5l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList()))).equals((new ArrayList<Long>(Arrays.asList())));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)1l, (long)2l, (long)3l, (long)4l, (long)5l, (long)6l, (long)7l, (long)8l)))).equals((new ArrayList<Long>(Arrays.asList((long)1l, (long)8l, (long)2l, (long)7l, (long)3l, (long)6l, (long)4l, (long)5l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)0l, (long)2l, (long)2l, (long)2l, (long)5l, (long)5l, (long)-5l, (long)-5l)))).equals((new ArrayList<Long>(Arrays.asList((long)-5l, (long)5l, (long)-5l, (long)5l, (long)0l, (long)2l, (long)2l, (long)2l))));
    strange_sort_list((new ArrayList<Long>(Arrays.asList((long)111111l)))).equals((new ArrayList<Long>(Arrays.asList((long)111111l))));
    }

}
